src/components/SearchResultList/SearchResultList.tsx

```diff
@@ -0,0 +1,113 @@
+}
+
+const SearchResultList = () => {
+  const [value, setValue] = React.useState<string>('');
+  const [pageNumber, setPageNumber] = React.useState<number>(DEFAULT_CURRENT_PAGE);
+  const [searchResults, setSearchResults] = React.useState<any[]>([]);
+  const [meta, setMeta] = React.useState<Meta>({
+    limit: PAGE_LIMIT,
+    totalCounts: 1,
+    currentPage: DEFAULT_CURRENT_PAGE,
+  });
+  const [loading, setLoading] = React.useState<boolean>(false);
+
+  const debouncedValue = useDebounce<string>(value, 500);
+
+  React.useEffect(() => {
+    const queryParams = {
+      maxRows: PAGE_LIMIT,
+      currentPage: pageNumber,
+    };
+
+    async function listSearchResults() {
+      setLoading(true);
+      try {
+        const data = await listSearchResultsAction(queryParams, debouncedValue);
+        const count = await getSearchResultCountAction(debouncedValue);
+
+        setSearchResults(data);
+        setMeta({ limit: PAGE_LIMIT, totalCounts: count, currentPage: pageNumber });
+        setLoading(false);
+      } catch (err: any) {
+        toast(err.message, 'error');
+        setLoading(false);
+      }
+    }
+
+    listSearchResults();
+  }, [pageNumber, debouncedValue]);
+
+  const handleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setValue(event.target.value);
+  };
+
+  return (
+    (!loading && (
+      <div className="list-searchresult">
+        <div className="list-searchresult__header">
+          <label htmlFor="table-search" className="sr-only">
+            Search
+          </label>
+          <div className="relative mt-1">
+            <input
+              onChange={handleChange}
+              type="text"
+              id="table-search"
+              value={value}
+              className="list-searchresult__input"
+              placeholder="Search for items"
+            />
+          </div>
+        </div>
+        <table className="table">
+          <thead className="table__head">
+            <tr>
+              <th className="table__head-column">Name</th>
+              <th className="table__head-column">Position</th>
+              <th className="table__head-column">Status</th>
+            </tr>
+          </thead>
+          <tbody data-testid="search-table__body">
+            {searchResults.map((result) => {
+              return (
+                <tr key={result.id} className="table__body-row">
+                  <td className="table__body-column">{result.keyword}</td>
+                  <td className="table__body-column">React Developer</td>
+                  <td className="table__body-column">{result.status}</td>
+                </tr>
+              );
+            })}
+          </tbody>
+        </table>
+        <Pagination
+          totalPageNumber={Math.ceil(meta.totalCounts / meta.limit)}
+          pageLimit={meta.limit}
+          pageNo={meta.currentPage}
+          handleChange={(number: number) => {
+            setPageNumber(number);
+          }}
```

Comment:
Will this still work? 👀 

```suggestion
          handleChange={setPageNumber}
```

Reply:
It is call back function. Meaning this number will be set from the pagination component to set the page number to query the database. It will not work.

Reply:
🤔 I still don't quite understand yet. I tried to run some tests on my local using both versions and it still work 🙈

Reply:
In this particular case. In the searchResult page the function on handleChange has no access to number of the page. So, a callback function is passed as a parameter to function. So, where the function is getting called they will pass the parameter. This is similar to the concept of withAuth that you have done in [With Auth for server actions](https://github.com/nimblehq/js-framework-benchmark/blob/develop/nextjs-13/src/lib/withAuth/withAuth.ts)